systemd: improve unit script drv naming

lheckemann opened the pull request:
###### Motivation for this change
I don't like having the nix store littered with derivations that all have the name "unit-script", containing nothing but `bin/useful-name-here`. This is a bit of an RFC since I'm not sure whether this might break any further things. I've run a few nixos tests with this change as a sanity check, but this could probably do with a run of all the tests and some more experienced eyes looking at it.

cc @grahamc since you expressed interest in the question

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

lheckemann commented:
@GrahamcOfBorg test gocd-server openldap firewall opensmtpd munin taskserver boot peerflix flannel initrd-network

GrahamcOfBorg commented:
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.43534&attempt_id=e7d06142-f296-4168-b858-eeaf51bed684)

Attempted: tests.gocd-server, tests.openldap, tests.firewall, tests.opensmtpd, tests.munin, tests.taskserver, tests.boot, tests.peerflix

The following builds were skipped because they don't evaluate on x86_64-linux: tests.flannel, tests.initrd-network

<details><summary>Partial log (click to expand)</summary><p>

```
error: the VM quit before connecting
the VM quit before connecting
cleaning up
killing server (pid 593)
killing client2 (pid 617)
killing client1 (pid 605)
vde_switch: EOF on stdin, cleaning up and exiting
vde_switch: Could not remove ctl dir '/build/vde1.ctl': Directory not empty
builder for '/nix/store/kkynl75230ighsj5j0bg2ldp7gqad153-vm-test-run-taskserver.drv' failed with exit code 255
error: build of '/nix/store/cic0ys2fdy3j3y4cv75c4rcbhr6ji58m-vm-test-run-opensmtpd.drv', '/nix/store/kkynl75230ighsj5j0bg2ldp7gqad153-vm-test-run-taskserver.drv' failed
```
</p></details>

lheckemann commented:
opensmtpd and taskserver tests worked fine here, so I'm guessing that's a fluke.

fpletz commented:
:+1: on the change in principle, but I wonder if we should keep the `unit-script` in the name for clarity.

lheckemann commented:
That's a possibility, although I think it's somewhat redundant given that the names already contain `-start` etc. Should I make the change?

globin commented:
:+1: for this, I'd also prefer having it prefixed with `unit-script`

lheckemann committed:
systemd: improve unit script drv naming

Also store scripts directly in the nix store rather than having the
superfluous /bin/ tree.

lheckemann commented:
done!

GrahamcOfBorg commented:
No attempt on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.43534&attempt_id=3634cb2a-b4dc-441a-a954-639a83eca46c)

The following builds were skipped because they don't evaluate on x86_64-darwin: systemd

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.43534&attempt_id=30261fca-8316-4b3d-9475-519bc64026a5)

Attempted: systemd

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/vlzj5l79c5m5m808s9axgnmy65vq04gf-systemd-238
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.43534&attempt_id=7e8d1b83-05a4-43f5-a137-d6f7b8ce47e0)

Attempted: systemd

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/3xgvh1fjc6p586425c3annc03xhzx9is-systemd-238
```
</p></details>

lheckemann commented:
@GrahamcOfBorg test gocd-server openldap firewall opensmtpd munin taskserver boot peerflix flannel initrd-network

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.43534&attempt_id=ef963aee-7cef-4a89-8910-597eebd86ff8)

Attempted: tests.openldap, tests.firewall, tests.opensmtpd, tests.taskserver, tests.boot, tests.peerflix

The following builds were skipped because they don't evaluate on aarch64-linux: tests.gocd-server, tests.munin, tests.flannel, tests.initrd-network

<details><summary>Partial log (click to expand)</summary><p>

```
building '/nix/store/1203g3jf2gsdb62ymz4lnjax4lgr3z1s-unit-nixos-upgrade.service.drv'...
building '/nix/store/5i7kiprwh9k2lp3hfdxah03py22x7id4-unit-nscd.service.drv'...
building '/nix/store/877psgdklhbn9aidi8dv44wq2paczlha-unit-nscd.service.drv'...
building '/nix/store/kizz7sv22pbnvzrg18fydak9b91p0p7i-unit-nscd.service.drv'...
building '/nix/store/wqpg2kgwv0b2965hwaq2y512ww417il8-unit-nscd.service.drv'...
building '/nix/store/zxw3w1bi8c57rq0wygycs5f1p1b745fm-unit-nscd.service.drv'...
building '/nix/store/b7p7n58y4bv4j7fv3axl8vclakrshvpm-unit-post-resume.service.drv'...
building '/nix/store/v5bjbj00n9651b5y3vg9629z5pq414cl-unit-post-resume.service.drv'...
error (ignored): aborting transaction: SQL logic error (in '/nix/var/nix/db/db.sqlite')
error: writing to file: No space left on device
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.43534&attempt_id=55f032ea-0045-47d2-b3b0-0e2181aca285)

Attempted: tests.gocd-server, tests.openldap, tests.firewall, tests.opensmtpd, tests.munin, tests.taskserver, tests.boot, tests.peerflix

The following builds were skipped because they don't evaluate on x86_64-linux: tests.flannel, tests.initrd-network

<details><summary>Partial log (click to expand)</summary><p>

```
client1: running command: sync
client1: exit status 0
10 out of 10 tests succeeded
test script finished in 46.09s
cleaning up
killing server (pid 600)
killing client2 (pid 624)
killing client1 (pid 612)
vde_switch: EOF on stdin, cleaning up and exiting
vde_switch: Could not remove ctl dir '/build/vde1.ctl': Directory not empty
```
</p></details>
